packages/liquidator/src/liquidationStrategy.js

```diff
@@ -142,46 +129,44 @@ module.exports = (
-          message:
-            "Liquidator bot is extending withdraw deadline, funds may need to be added to liquidate full position",
-          position: position,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          syntheticTokenBalance: syntheticTokenBalance.toString(),
-          maxTokensToLiquidateWei: maxTokensToLiquidateWei ? maxTokensToLiquidateWei.toString() : null,
-          whaleDefenseFundWei: whaleDefenseFundWei.toString(),
-          currentBlockTime,
-          ...logInfo
-        });
-        // this assume empMinSponsorSize will always be <= maxTokensToLiquidate
-        return createLiquidationParams({
-          sponsor: position.sponsor,
-          // Only liquidating the minimum size to extend withdraw
-          tokensToLiquidate: minSponsorSize,
-          currentBlockTime,
-          maxCollateralPerToken
-        });
-      }
-    }
+    let tokensToLiquidate = toBN(0);
 
-    // WDF strategy not activated, calculate tokens to liquidate respecting all constraints
-    const tokensToLiquidate = calculateTokensToLiquidate({
-      syntheticTokenBalance,
-      positionTokens: position.numTokens,
-      maxTokensToLiquidateWei,
-      withdrawRequestPending: hasWithdrawRequestPending({ position, currentBlockTime })
-    });
+    if (!wdfActive || !hasWithdrawRequestPending({ position, currentBlockTime })) {
+      // 1) If either WDF is not activated or there is no withdraw pending, then
+      //    submit liquidation using all capital
+      //     - This will include withdraws that have expired already.
+      tokensToLiquidate = calculateTokensToLiquidate({
+        syntheticTokenBalance,
+        positionTokens: position.numTokens,
+        maxTokensToLiquidateWei
+      });
+    } else if (canLiquidateFully({ position, syntheticTokenBalance })) {
+      // 2) If you can fully liquidate, do it, regardless of whether WDF is activated.
+      tokensToLiquidate = calculateTokensToLiquidate({
+        syntheticTokenBalance,
+        positionTokens: position.numTokens,
+        maxTokensToLiquidateWei
+      });
+    } else if (
+      wdfActive &&
+      hasWithdrawRequestPending({ position, currentBlockTime }) &&
+      passedDefenseActivationPercent({ position, currentBlockTime }) &&
+      canLiquidateMinimum({ position, syntheticTokenBalance })
+    ) {
+      // 3) If WDF is active and withdraw has passed liveness, then submit min liquidation
+      emit("info", {
+        message: "Liquidator bot is extending withdraw deadline, funds may need to be added to liquidate full position",
+        position: position,
+        maxLiquidationPrice: maxCollateralPerToken.toString(),
+        syntheticTokenBalance: syntheticTokenBalance.toString(),
+        maxTokensToLiquidateWei: maxTokensToLiquidateWei ? maxTokensToLiquidateWei.toString() : null,
+        currentBlockTime,
+        ...logInfo
+      });
+      tokensToLiquidate = minSponsorSize;
+    } else {
+      // 4) Do nothing.
```

Comment:
just trying to follow the logic, this would be a case where:
- wdf not active and there's a pending withdraw request and we cant liquidate for some reason
- wdf active and there's no pending withdraw request and we cant liquidate for some reason
-
I think its probably right, and in this case we return nothing from this function which might trigger a log in the liquidator. Is that ok and does that log mean anything to us?

Reply:
> wdf not active and there's a pending withdraw request and we cant liquidate for some reason
If "wdf is not active", then the first case gets triggered `if (!wdfActive || !hasWithdrawRequestPending({ position, currentBlockTime }))` and the liquidator will use whatever balance it has.

> wdf active and there's no pending withdraw request and we cant liquidate for some reason
If the "wdf is active", and there is pending withdrawal, then the first case also gets triggered because of the RHS. This fourth case occurs where the wdf is active AND the liquidator cannot fully liquidate a position BUT either the `defensePassedActivationPercent` has not triggered or the liquidator doesn't have enough balance to submit a min liquidation.

Does that make sense?

> I think its probably right, and in this case we return nothing from this function which might trigger a log in the liquidator. Is that ok and does that log mean anything to us?

Yes right now we actually try to parse this situation in `liquidator` and send a cutomized log in this [section](https://github.com/UMAprotocol/protocol/blob/master/packages/liquidator/src/liquidator.js#L300)

Reply:
yea that makes a lot of sense. thanks for breaking it down. Also looked at logs and that looks good to me

Reply:
I do think we should move all the log logic to the `liquidatorStrategy` rather than reading it from `liquidator` but I'll save for another PR and curious what other eng's think

Reply:
i think its reasonable to do it in a later PR if needed

Reply:
Actually, I'm going to implement this change, see below